build-plugin.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+#
+# Building migrations with datastore plugin
+#
+# This script builds migrations with a datastore plugin.  Specify the datastore
+# plugin repo and one or more migrations to build.  Each migration binary is
+# built in its module subdirectory.  Run a migration binary directly, or copy
+# it into a directory in PATH to be run by ipfs-update or fs-repo-migrations.
+#
+# Example:
+# ./build-plugin.sh github.com/ipfs/go-ds-s3 10-to-11 11-to-12
+#
+set -eou pipefail
+
+function usage() {
+    echo "usage: $0 plugin_repo x-to-y ...">&2
+    echo "example: $0 github.com/ipfs/go-ds-s3 10-to-11" >&2
```

Comment:
Two feature requests I'd add here, that can happen after we merge this initial one but would be nice to have are:
- Being able to choose the version of the repo not just the repo itself (e.g. I might need to build an old migration against an old version)
- Being able to add multiple plugins to a given binary

Reply:
- A user can now specify the version of the migration by following it with `@<version_or_hash>`
- The script builds a single migration, with one or more plugins.

Example:
```sh
./build-plugin.sh 10-to-11 github.com/ipfs/go-ds-s3 github.com/ipfs/go-ds-swift@v0.1.0
```

Reply:
Awesome, looking forward to giving this a spin.